dali/operators/reader/loader/webdataset/tar_utils.cc

```diff
@@ -140,15 +139,32 @@ bool TarArchive::NextFile() {
   if (eof_) {
     return false;
   }
-  Skip(RoundToBlockSize(filesize_) - readoffset_);
-  eof_ = ParseHeader();
+
+  if (stream_->Tell() + RoundToBlockSize(filesize_) - readoffset_ >= stream_->Size()) {
```

Comment:
If there's some inconsistency between Tell, filesize_ and readoffset_ and the left-hand-side goes negative, this will still be true (due to the value being coerced to unsigned and wrapping) and will trigger EOF - shouldn't it rather be an error?
```suggestion
  int64_t offset = stream_->Tell() + RoundToBlockSize(filesize_) - readoffset;
  assert(offset >= 0);
  if (static_cast<size_t>(offset) >= stream_->Size()) {
```